torch/csrc/distributed/c10d/init.cpp

```diff
@@ -87,11 +87,13 @@ PyObject* c10d_init(PyObject* _unused) {
               std::vector<std::vector<torch::autograd::Variable>>,
               std::vector<std::vector<size_t>>,
               std::shared_ptr<::c10d::ProcessGroup>,
-              std::vector<std::vector<bool>>>(),
+              std::vector<std::vector<bool>>,
+              bool>(),
           py::arg("replicas"),
           py::arg("bucket_indices"),
           py::arg("process_group"),
-          py::arg("expect_sparse_gradients") = std::vector<std::vector<bool>>())
+          py::arg("expect_sparse_gradients") = std::vector<std::vector<bool>>(),
+          py::arg("delay_allreduce") = false)
```

Comment:
There is `torch.distributed.all_reduce` instead of `allreduce` for the term "all-reduce". I suggest renaming to `delay_all_reduce` for consistency with already existing APIs.